### PR TITLE
Speed up CI

### PR DESCRIPTION
**Pull request by asaj**
### Description

Speeds up CI by:
- fixing node_modules caching
- sending fewer messages in the e2e test
- failing less frequently in the e2e test
- removing yarn build cache from e2e test which always misses 

### Drive-by changes

- Update hyperlane/token to latest main
- Fix coverage-sol to read proper files
- Never cancel in-progress e2e tests on main

### CODE REVIEW

File: .github/workflows/e2e.yml [hunk 1]
```diff
@@ -60,4 +62,4 @@ jobs:
         env:
           E2E_CI_MODE: 'true'
           E2E_CI_TIMEOUT_SEC: '600'
-          E2E_KATHY_ROUNDS: '3'
```
Comment (mattiecnvr) [inline .github/workflows/e2e.yml:63]: I don't think we should need this change with the change to frequency in failures.

**Review by mattiecnvr — APPROVED** [review]
Did you mean to commit changes to the token submodule?

**Comment by asaj** [comment]
@mattiecnvr yes I did mean to commit that, see the PR description drivebys

File: .github/workflows/e2e.yml (hunk 1 above)
Comment (asaj) [reply .github/workflows/e2e.yml:63]: We don't *need* it, but it's still better IMO. I'm not sure what 3 rounds would catch that 2 won't, and 2 is faster than 3 :)

**Pull request merged**